plugin-maven/src/main/java/com/diffplug/spotless/maven/GitRatchetMaven.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.maven;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.function.Predicate;
+
+import org.eclipse.jgit.lib.ObjectId;
+
+import com.diffplug.common.base.Errors;
+import com.diffplug.spotless.extra.GitRatchet;
+
+class GitRatchetMaven extends GitRatchet<File> {
```

Comment:
This class looks like a singleton. Maybe make it final and add a private constructor?

Reply:
Thanks, good call, will do!